src/overlays/actors/ovl_En_Cne_01/z_en_cne_01.c

```diff
@@ -33,85 +34,295 @@ const ActorInit En_Cne_01_InitVars = {
+    talkstate = Message_GetState(&globalCtx->msgCtx);
+    this->inMsgState3 = (talkstate == 3) ? true : false;
+
+    switch (talkstate) {
+        case 0:
+            yaw = ABS_ALT(this->actor.shape.rot.y - this->actor.yawTowardsPlayer);
+            if (yaw < 0x64) {
+                func_801518B0(globalCtx, this->textId, NULL);
+            }
+            break;
+        case 2:
+            this->actor.textId = 0;
+            this->focusTarget = this->tmpFocusTarget;
+            this->headRot = this->tmpHeadRot;
+            this->torsoRot = this->tmpTorsoRot;
+            this->actor.shape.rot.y = this->actor.world.rot.y;
+            this->actionFunc = this->tmpActionFunc;
+            this->tmpActionFunc = NULL;
+            break;
+    }
+}
+
+void EnCne01_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnCne01* this = THIS;
+
+    this->enHy.animObjIndex = func_8013D924(OBJECT_OS_ANIME, globalCtx);
+    this->enHy.headObjIndex = func_8013D924(OBJECT_CNE, globalCtx);
+    this->enHy.skelUpperObjIndex = func_8013D924(OBJECT_CNE, globalCtx);
+    this->enHy.skelLowerObjIndex = func_8013D924(OBJECT_CNE, globalCtx);
+
+    if ((this->enHy.animObjIndex < 0) || (this->enHy.headObjIndex < 0) || (this->enHy.skelUpperObjIndex < 0) ||
+        (this->enHy.skelLowerObjIndex < 0)) {
+        Actor_MarkForDeath(&this->enHy.actor);
+    }
+    this->enHy.actor.draw = NULL;
+    Collider_InitCylinder(globalCtx, &this->enHy.collider);
+    Collider_SetCylinder(globalCtx, &this->enHy.collider, &this->enHy.actor, &sCylinderInit);
+    CollisionCheck_SetInfo2(&this->enHy.actor.colChkInfo, &sDamageTable, &sColChkInfoInit);
+    this->enHy.actor.flags &= ~1;
+    this->enHy.path = func_8013D648(globalCtx, ENCNE01_GET_PATH(&this->enHy.actor), ENCNE01_NO_PATH);
+    this->enHy.waitingOnInit = true;
+    Actor_SetScale(&this->enHy.actor, 0.01f);
+    this->enHy.actionFunc = EnCne01_FinishInit;
+}
+
+void EnCne01_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnCne01* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->enHy.collider);
+}
+
+void EnCne01_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnCne01* this = THIS;
+
+    EnCne01_TestIsTalking(this, globalCtx);
+    this->enHy.actionFunc(&this->enHy, globalCtx);
+    Actor_UpdateBgCheckInfo(globalCtx, &this->enHy.actor, 0.0f, 0.0f, 0.0f, 4);
+    EnCne01_UpdateModel(this, globalCtx);
+    func_809CB4A0(this, globalCtx);
+}
 
-extern ColliderCylinderInit D_809CBF00;
-extern CollisionCheckInfoInit2 D_809CBF2C;
-extern DamageTable D_809CBF38;
+s32 EnCne01_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
+                             Actor* thisx) {
+    EnCne01* this = THIS;
+    s8 bodyPart;
+    Vec3f zeroVec = { 0.0f, 0.0f, 0.0f };
 
-extern UNK_TYPE D_060000F0;
-extern UNK_TYPE D_06001300;
+    bodyPart = gEnHyBodyParts[limbIndex];
+    if (bodyPart >= 0) {
```

Comment:
Could this be `bodyPart >= CNE_LIMB_NONE`?

Reply:
It doesn't quite make sense, as these are bodyParts not limbs, though they do have a link to each limb based on `gEnHyBodyParts`. The bodyParts could probably have their own enum, but that should probably be discussed based on all the actors that use them.